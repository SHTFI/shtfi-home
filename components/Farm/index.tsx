export { default as FarmIntroSection } from "./IntroSection";
export { default as FarmSelectSection } from "./FarmSelectSection";
