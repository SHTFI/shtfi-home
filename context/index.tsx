export { default as Web3Providers } from "./web3Providers";
export * from "./web3";
