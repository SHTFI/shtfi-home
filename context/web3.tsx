import { Web3Provider } from "@ethersproject/providers";
import { InjectedConnector } from "@web3-react/injected-connector";
import { DEFAULT_CHAIN_ID, SUPPORTED_CHAIN_IDS } from "config";
import { NetworkConnector } from "connectors/networkConnector";

enum ConnectorNames {
  Injected = "Injected",
  Network = "Network",
}

type GetConnectorsByName = {
  [ConnectorName in ConnectorNames]: InjectedConnector | NetworkConnector;
};

// Set an object with our node urls
const nodeUrls = {
  [97 as number]: [
    "https://data-seed-prebsc-1-s1.binance.org:8545",
    "https://data-seed-prebsc-2-s2.binance.org:8545",
  ],
  [56 as number]: [
    "https://bsc-dataseed.binance.org/",
    "https://bsc-dataseed1.defibit.io/",
  ],
};
// Make an injected web3 connector to connect to meta mask
export const injectedWeb3 = new InjectedConnector({
  supportedChainIds: SUPPORTED_CHAIN_IDS,
});
// Make a network connector to allow us to show blockchain data without the user bringing meta mask
export const networkWeb3 = new NetworkConnector({
  urls: SUPPORTED_CHAIN_IDS.reduce((op: any, id: number) => {
    op[id] = nodeUrls[id][0];
    return op;
  }, {}),
  defaultChainId: DEFAULT_CHAIN_ID,
});
// Export an object which allows us to get the connector by name
export const getConnectorsByName: GetConnectorsByName = {
  [ConnectorNames.Injected]: injectedWeb3,
  [ConnectorNames.Network]: networkWeb3,
};
// Export a function to allow us to easily get our web3 library
export const getWeb3Library = (provider: any): Web3Provider =>
  new Web3Provider(provider);
