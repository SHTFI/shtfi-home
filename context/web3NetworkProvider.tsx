import { createWeb3ReactRoot } from "@web3-react/core";
const DefaultWeb3NetworkProvider = createWeb3ReactRoot("Network");

const DefaultWeb3NetworkProviderSSR: React.FC<{ getLibrary: any }> = ({
  children,
  getLibrary,
}) => {
  return (
    <DefaultWeb3NetworkProvider getLibrary={getLibrary}>
      {children}
    </DefaultWeb3NetworkProvider>
  );
};

export default DefaultWeb3NetworkProviderSSR;
