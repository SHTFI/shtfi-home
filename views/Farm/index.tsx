export { default as FarmIntroSection } from "./IntroSection";
export { default as FarmMeta } from "./Meta";
export { default as FarmSelectSection } from "./FarmSelectSection";
