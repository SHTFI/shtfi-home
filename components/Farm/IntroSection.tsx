import styled from "styled-components";
import { Heading } from "components";
const FarmIntroSection: React.FC = () => {
  return (
    <StyledIntroSection>
      <Heading level={2} text="SHTFI FARMS" />
      <p>
        Use the input field below to search for a SHTFI farm. Please remember
        that the smart contracts which govern these farms have been made as a
        side project and that you use them at your own risk.
      </p>
    </StyledIntroSection>
  );
};
const StyledIntroSection = styled.section`
  padding: var(--large-space);
`;
export default FarmIntroSection;
