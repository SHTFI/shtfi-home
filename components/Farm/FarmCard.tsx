import Image from "next/image";
import styled from "styled-components";
import { Card, Button } from "components";
import { useFarmerInfo, useFarmInfo } from "hooks";
import { useWeb3 } from "hooks";
import { isMetaMaskInstalled } from "utils";
import { injectedWeb3 } from "context";
import { FarmCardProps } from "types";

const FarmCard: React.FC<FarmCardProps> = ({
  stakedToken,
  rewardToken,
  farmAddress,
}) => {
  // Get account from web3 hook
  const { account, library, active, activate } = useWeb3();
  const { farmInfo } = useFarmInfo(farmAddress, stakedToken.contract[97]);
  const { farmerInfo } = useFarmerInfo(
    account,
    farmInfo?.address,
    farmInfo?.id
  );

  if (!!!farmInfo || !!!farmerInfo) return null;

  const connectToMetaMask = (active: boolean) => {
    if (window.ethereum.isMetaMask) {
      activate(injectedWeb3, undefined, true);
    }
  };
  return (
    <StyledFarmCard>
      <StyledFarmHeader>
        <StyledFarmHeaderIcons>
          <StyledTokenIcon>
            <Image
              src={rewardToken.icon}
              alt={`${rewardToken.name} icon`}
              width="75px"
              height="75px"
            />
          </StyledTokenIcon>
          <StyledTokenIcon>
            <Image
              src={stakedToken.icon}
              alt={`${stakedToken.name} icon`}
              width="50px"
              height="50px"
            />
          </StyledTokenIcon>
        </StyledFarmHeaderIcons>
        <StyledFarmHeaderMeta>
          <p>
            Stake: <span>{stakedToken.ticker}</span>
          </p>
          <p>
            Get: <span>{rewardToken.ticker}</span>
          </p>
        </StyledFarmHeaderMeta>
      </StyledFarmHeader>
      <StyledFarmInfo>
        <p>
          Total {stakedToken.ticker}: <span>{farmInfo?.stakedBalance}</span>
        </p>
        <p>
          {rewardToken.ticker} per block:{" "}
          <span>~{farmInfo?.rewardPerBlock}</span>
        </p>
        <p>
          {rewardToken.ticker} per {stakedToken.ticker}:{" "}
          <span>~{farmInfo.rewardPerStake}</span>
        </p>
      </StyledFarmInfo>

      <StyledFarmActions data-connected={!!account && active}>
        <StyledFarmAction aria-hidden={!account}>
          <StyledFarmActionInfo>
            <span>Your Stake:</span>
            <span>
              {farmerInfo.stakedBalance} {stakedToken.ticker}
            </span>
          </StyledFarmActionInfo>
          <StyledFarmActionButton
            id={`unstake-${stakedToken.ticker}-${rewardToken.ticker}`}
            size="med"
          >
            Unstake
          </StyledFarmActionButton>
        </StyledFarmAction>
        <StyledFarmAction aria-hidden={!account}>
          <StyledFarmActionInfo>
            <span>Your Rewards:</span>
            <span>
              ~{farmerInfo.pendingRewards} {rewardToken.ticker}
            </span>
          </StyledFarmActionInfo>
          <StyledFarmActionButton
            id={`claim-${stakedToken.ticker}-${rewardToken.ticker}`}
            size="med"
          >
            Claim
          </StyledFarmActionButton>
        </StyledFarmAction>
        <StyledFarmAction justify="center" aria-hidden={!account}>
          <StyledFarmActionButton
            id={`stake-${stakedToken.ticker}-${rewardToken.ticker}`}
            size="med"
          >
            Stake
          </StyledFarmActionButton>
        </StyledFarmAction>
        <StyledFarmAction data-login aria-hidden={!!account}>
          <StyledFarmActionButton
            id={`login-${stakedToken.ticker}-${rewardToken.ticker}`}
            size="large"
            onClick={() => connectToMetaMask(active)}
          >
            {isMetaMaskInstalled() ? "Connect to MetaMask" : "Install MetaMask"}
          </StyledFarmActionButton>
        </StyledFarmAction>
      </StyledFarmActions>
    </StyledFarmCard>
  );
};

const StyledFarmCard = styled(Card)`
  border: 3px solid var(--blue);
  margin: var(--large-space);
  border-radius: calc(var(--large-space) * 3);
`;

const StyledFarmHeader = styled.header`
  display: flex;
  justify-content: center;
  padding: calc(var(--large-space) * 2) var(--large-space);
`;

const StyledFarmHeaderIcons = styled.div`
  display: inline-block;
  position: relative;
  width: 81.25px;
  height: 62.5px;
  margin-right: var(--large-space);
`;

const StyledTokenIcon = styled.div`
  position: absolute;
  display: inline-block;
  border-radius: 50%;
  overflow: hidden;

  &:first-child {
    left: 0;
    top: 0;
    width: 62.5px;
    height: 62.5px;
  }

  &:nth-child(2) {
    left: 31.25px;
    top: 6.25px;
    width: 50px;
    height: 50px;
  }
`;

const StyledFarmHeaderMeta = styled.div`
  align-self: center;

  p {
    > span {
      font-weight: bold;
    }
  }
`;

const StyledFarmInfo = styled.div`
  font-size: var(--meta-font);
  font-weight: bold;
  margin: var(--large-space);
  > p {
    > span {
      font-weight: normal;
    }
  }
`;

const StyledFarmActions = styled.div`
  position: relative;
  &[data-connected="true"] {
    > div {
      transition: opacity 0.3s ease 0s, visibility 0.1s ease 0.3s;
    }
    > div:not([data-login]) {
      opacity: 1;
      visibility: visible;
    }
    > div[data-login] {
      opacity: 0;
      visibility: hidden;
    }
  }
  &[data-connected="false"] {
    > div {
      transition: opacity 0.3s ease 0.1s, visibility 0.1s ease 0s;
    }
    > div:not([data-login]) {
      opacity: 0;
      visibility: hidden;
    }

    > div[data-login] {
      opacity: 1;
      visibility: visible;
    }
  }
`;

const StyledFarmAction = styled.div<{ justify?: string }>`
  display: flex;
  margin: var(--large-space);

  &:last-of-type {
    margin-top: calc(var(--large-space) * 2);
  }

  ${(props) => {
    const { justify } = props;
    return `
      justify-content: ${!!justify ? justify : "space-between"};
    `;
  }}

  &[data-login] {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    margin: 0;
  }
`;
const StyledFarmActionInfo = styled.div`
  align-self: center;
  > span {
    display: block;
  }

  > span:first-child {
    font-size: var(--meta-font);
  }
  > span:nth-child(2) {
    font-size: var(--reg-font);
    font-weight: bold;
    margin-right: auto;
  }
`;

const StyledFarmActionButton = styled(Button)`
  align-self: center;
`;

export default FarmCard;
