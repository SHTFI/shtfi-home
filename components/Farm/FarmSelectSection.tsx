import { Input } from "components";
import { useWeb3 } from "hooks";
import styled from "styled-components";
import FarmCard from "./FarmCard";

const FarmSelectSection: React.FC<FarmSectionProps> = ({ farms, ...rest }) => {
  const { active, library, account } = useWeb3();
  if (active && account) {
    console.log(library?.getBlockNumber());
  }
  return (
    <StyledFarmSection {...rest}>
      <Input placeholder="Search for coin" button={true} />
      <StyledFarmCards>
        {farms.map(({ stakedToken, rewardToken }, i) => (
          <FarmCard
            key={`${stakedToken.ticker}-${rewardToken.ticker}-card-${i}`}
            stakedToken={stakedToken}
            rewardToken={rewardToken}
          />
        ))}
      </StyledFarmCards>
    </StyledFarmSection>
  );
};

const StyledFarmSection = styled.section`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;
const StyledFarmCards = styled.div`
  display: flex;
  flex-wrap: wrap;
  width: 100%;
  justify-content: center;
  padding: calc(var(--large-space) * 2) 0;
`;

export default FarmSelectSection;
