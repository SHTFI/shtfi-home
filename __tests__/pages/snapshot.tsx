import React from "react";
import renderer from "react-test-renderer";
import Index from "pages/index";
import Farm from "pages/farm";
import preloadAll from "jest-next-dynamic";
import { Web3Providers } from "context";

describe("<Index />", () => {
  it("renders homepage unchanged", () => {
    const tree = renderer.create(<Index />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("<Farm />", () => {
  beforeAll(async () => {
    await preloadAll();
  });
  it("renders farm unchanged", () => {
    const tree = renderer
      .create(
        <Web3Providers>
          <Farm />
        </Web3Providers>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
