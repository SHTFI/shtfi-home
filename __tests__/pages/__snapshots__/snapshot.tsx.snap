// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Farm /> renders farm unchanged 1`] = `null`;

exports[`<Index /> renders homepage unchanged 1`] = `
<div
  className="pages__StyledWrapper-sc-tddur0-0 tYBfw"
>
  <main
    className="pages__StyledMain-sc-tddur0-1"
  >
    <header
      className="Header__StyledHeader-sc-k859ru-0 iRENco"
    >
      <picture
        className="Header__StyledHeaderLogo-sc-k859ru-1 iRvyht"
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "inline-block",
              "margin": 0,
              "maxWidth": "100%",
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "display": "block",
                "maxWidth": "100%",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTI1IiBoZWlnaHQ9IjEyNSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
              style={
                Object {
                  "border": "none",
                  "display": "block",
                  "margin": 0,
                  "maxWidth": "100%",
                  "padding": 0,
                }
              }
            />
          </div>
          <img
            alt="SHTFI logo"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              alt="SHTFI logo"
              data-nimg="intrinsic"
              decoding="async"
              loading="lazy"
              src="/_next/image?url=%2Fassets%2Ficons%2Fshtfi%2Fshtfi-bg.svg&w=256&q=75"
              srcSet="/_next/image?url=%2Fassets%2Ficons%2Fshtfi%2Fshtfi-bg.svg&w=128&q=75 1x, /_next/image?url=%2Fassets%2Ficons%2Fshtfi%2Fshtfi-bg.svg&w=256&q=75 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </div>
      </picture>
      <div
        className="Header__StyledHeaderContent-sc-k859ru-2 bonaFM"
      >
        <h1>
          SHTFI
        </h1>
        <p>
          Open Source Yield Farming and Defi Protocol
        </p>
        <div
          className="SocialLinks__StyledSocialLinks-sc-1ts488y-0 fxCZUI"
        >
          <a
            className="SocialLinks__StyledSocialLink-sc-1ts488y-1 ikQGNt"
            href="https://github.com/SHTFI"
            rel="noopener noreferrer"
            target="_blank"
            title="View SHTFI on github"
          >
            <svg
              color="white"
              fill="currentColor"
              height={20}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": "white",
                }
              }
              viewBox="0 0 496 512"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
              />
            </svg>
          </a>
          <a
            className="SocialLinks__StyledSocialLink-sc-1ts488y-1 ikQGNt"
            href="https://twitter.com/shtfio"
            rel="noopener noreferrer"
            target="_blank"
            title="View SHTFI on twitter"
          >
            <svg
              color="white"
              fill="currentColor"
              height={20}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": "white",
                }
              }
              viewBox="0 0 512 512"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
              />
            </svg>
          </a>
        </div>
      </div>
    </header>
    <section
      className="IntroSection__StyledIntroSection-sc-1e6ymzb-0 hQoFC"
      id="intro"
    >
      <h2
        className="Heading__H2-sc-1n3vzci-1 eSuOyG"
      >
        Yield Farming Like Never Before
      </h2>
      <p>
        SHTFI is a brand new yield farming and defi protocol with a unique distribution mechanism which will see 0 pre-mine and 0 dev fund.
      </p>
    </section>
    <section
      className="Tokenomics__StyledTokenomics-sc-12yavak-0 gNAiJh"
      id="tokenomics"
    >
      <div
        className="Tokenomics__StyledTokenomicsContent-sc-12yavak-1 hNQFxb"
      >
        <h2
          className="Heading__H2-sc-1n3vzci-1 eSuOyG"
        >
          Tokenomics
        </h2>
        <div
          className="Tokenomics__StyledTokenomicsData-sc-12yavak-2 bRGlBE"
          data-large="true"
        >
          <span>
            0
          </span>
          <h4>
            Dev Fund
          </h4>
        </div>
        <div
          className="Tokenomics__StyledTokenomicsData-sc-12yavak-2 bRGlBE"
          data-large="true"
        >
          <span>
            0
          </span>
          <h4>
            Pre Mine
          </h4>
        </div>
        <div
          className="Tokenomics__StyledTokenomicsData-sc-12yavak-2 bRGlBE"
          data-large="true"
        >
          <span>
            0
          </span>
          <h4>
            Marketing
          </h4>
        </div>
        <div
          className="Tokenomics__StyledTokenomicsData-sc-12yavak-2 bRGlBE"
          data-large="true"
        >
          <span>
            222,222
          </span>
          <h4>
            Max Supply
          </h4>
        </div>
        <div
          className="Tokenomics__StyledTokenomicsData-sc-12yavak-2 bRGlBE"
          data-large="true"
        >
          <span>
            100%
          </span>
          <h4>
            SHTFI Farm
          </h4>
        </div>
      </div>
    </section>
    <section
      className="FeaturesSection__StyledFeaturesSection-sc-1rxht1h-0 ikjAoG"
    >
      <div
        className="FeaturesSection__StyledCardSection-sc-1rxht1h-1 gMSnZs"
        id="features"
      >
        <div
          className="Card__StyledCard-sc-19xhe45-0 bfdsLg"
        >
          <div
            className="FeaturesSection__StyledCardContent-sc-1rxht1h-2 hdTAGf"
          >
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "margin": 0,
                  "maxWidth": "100%",
                  "overflow": "hidden",
                  "position": "relative",
                }
              }
            >
              <div
                style={
                  Object {
                    "boxSizing": "border-box",
                    "display": "block",
                    "maxWidth": "100%",
                  }
                }
              >
                <img
                  alt=""
                  aria-hidden={true}
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAwIiBoZWlnaHQ9IjEwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                  style={
                    Object {
                      "border": "none",
                      "display": "block",
                      "margin": 0,
                      "maxWidth": "100%",
                      "padding": 0,
                    }
                  }
                />
              </div>
              <img
                alt="SHTFI toilet roll"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
              <noscript>
                <img
                  alt="SHTFI toilet roll"
                  data-nimg="intrinsic"
                  decoding="async"
                  loading="lazy"
                  src="/_next/image?url=%2Fassets%2Ficons%2Fshtfi%2Ftoilet-roll.svg&w=256&q=75"
                  srcSet="/_next/image?url=%2Fassets%2Ficons%2Fshtfi%2Ftoilet-roll.svg&w=128&q=75 1x, /_next/image?url=%2Fassets%2Ficons%2Fshtfi%2Ftoilet-roll.svg&w=256&q=75 2x"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </noscript>
            </div>
            <h3
              className="Heading__H3-sc-1n3vzci-2 jNenAk"
              style={
                Object {
                  "marginTop": 0,
                }
              }
            >
              NFTs? Why not!?
            </h3>
            <p>
              The SHTFI protocol will not be without its own set of useless NFTs which, in reality, mean nothing whatsoever. That doesnt mean you cant still tell yourself you own a unique picture of poop which no-one else has!
            </p>
            <button
              className="Button__StyledButton-sc-9q0s2c-0 iYdTtK"
              color="blue"
              data-type="modal-open"
              onClick={[Function]}
              size="large"
            >
              But, Why?
            </button>
            <div
              aria-hidden={true}
              className="Modal__StyledModalWrapper-sc-19pkxm0-0 bjigMf"
              data-type="wrapper"
              onClick={[Function]}
            >
              <span
                className="Modal__StyledUnderlay-sc-19pkxm0-1 flsNfz"
                data-type="underlay"
              />
              <aside
                aria-hidden={true}
                className="Modal__StyledModal-sc-19pkxm0-2 eufCoz"
              >
                <div>
                  <h4
                    className="Heading__H4-sc-1n3vzci-3 crxgEm"
                  >
                    Because we can!
                  </h4>
                  <p>
                    NFTs have become all the rage for reasons that are beyond the point of fungibility and the SHTFI protocol is all about being pointless. So lets do it! Lets make some NFTs!
                  </p>
                </div>
                <button
                  className="Button__StyledButton-sc-9q0s2c-0 dawuiH"
                  color="blue"
                  data-type="modal-close"
                  onClick={[Function]}
                  size="med"
                >
                  Close
                </button>
              </aside>
            </div>
          </div>
        </div>
      </div>
      <div
        className="FeaturesSection__StyledSliderSection-sc-1rxht1h-3 dxKhux"
        id="about"
      >
        <h2
          className="Heading__H2-sc-1n3vzci-1 eSuOyG"
        >
          What is SHTFI?
        </h2>
        <div
          className="Slider__StyledSlider-sc-xmb16q-0 hTpsgu"
        >
          <div
            className="Slider__StyledSlides-sc-xmb16q-1 cKWpYB"
          >
            <div
              className="Slider__StyledSlide-sc-xmb16q-2"
            >
              <div>
                <div
                  className="TextIcon__StyledWrapper-sc-17ldqsc-0 cMAoTk"
                >
                  <svg
                    fill="currentColor"
                    height={50}
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 512 512"
                    width={50}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M379.42 24.066l-28.059 87.407 28.268 5.941-3.098 15.352-52.25 47.843-51.5-43.125-23.404-4.093c8.217-14.33 14.683-32.77 16.404-49.594 10.02-2.28 34.92-5.675 46.094-10.059-4.997-10.285-30.197-16.906-48.7-16.316-1.733-20.713-8.88-29.054-34.155-27.902-25.276 1.151-32.972 6.601-30.16 36.423-18.866 4.127-38.097 12.616-39.74 27.084 7.87-.307 32.96-2.896 40.724-3.011.66 14.1 4.4 27.847 9.97 36.375l-35.158-6.125L106 195.922l77.344 55.875 1.625 16.844-34.19 215.75h38.375l38.315-169.25 47.873 169.25h37.47l-3.564-16.407 17.094 16.407 63.062-322.532c5.01-4.54 9.265-8.481 12.094-11.312.177-10.537-2.537-18.942-5.094-24.5l.971-4.902 27.238 5.724 8.444-93.117-22.846 68.781-10.848-2.256 6.635-72.658-21.568 69.55-11.217-2.333 6.207-70.77zm26.578 3.553l.002-.01h-.002v.01zM182.844 153.39l.344 64.095-31.5-23.75 31.156-40.345zm88.031 21.252l50.875 45.937s22.993-19.456 44.875-38.531l-57.438 285.562-42.812-197.529 4.5-95.44z"
                    />
                  </svg>
                  <h4
                    className="TextIcon__StyledHeading-sc-17ldqsc-1 bUGUqr"
                  >
                    Crypto Farm
                  </h4>
                  <p
                    className="TextIcon__StyledCopy-sc-17ldqsc-2 dkqrcv"
                  >
                    The most open and, probably, shit crypto farm of all time. Our smart contract was NOT forked from another major project and, as a result, is probably full of vulnerabilities. 
                  </p>
                </div>
              </div>
            </div>
          </div>
          <nav
            className="Slider__StyledNav-sc-xmb16q-3 eUNoA-D"
          >
            <button
              className="Button__StyledButton-sc-9q0s2c-0 iYdTtK"
              color="blue"
              disabled={true}
              onClick={[Function]}
              size="large"
            >
              &lt;
            </button>
            <button
              className="Button__StyledButton-sc-9q0s2c-0 iYdTtK"
              color="blue"
              disabled={false}
              onClick={[Function]}
              size="large"
            >
              &gt;
            </button>
          </nav>
        </div>
      </div>
    </section>
  </main>
</div>
`;
